### PR TITLE
MSVC Compatibility Changes 

### PR DESCRIPTION
**Pull request by jamiecook**
This PR addresses two issues: 
- [x] Remove usage of GCC specific compiler macro which was an issue while compiling on MSVC 2019 and other MSVC compilers.
- [x] Avoid use of designated initializer sytnax which is only supported in MSVC 2019+ and only when c++20 standard is enabled.

### CODE REVIEW

File: tensorflow/lite/delegates/external/external_delegate.cc [hunk 1]
```diff
@@ -155,12 +155,12 @@ ExternalDelegateWrapper::ExternalDelegateWrapper(
...
-          .CopyFromBufferHandle = nullptr,
-          .CopyToBufferHandle = nullptr,
-          .FreeBufferHandle = nullptr,
-          .flags = external_delegate_->flags,
+          reinterpret_cast<void*>(this),    // .data =
+          DelegatePrepare,                  // .Prepare =
+          nullptr,                          // .CopyFromBufferHandle =
+          nullptr,                          // .CopyToBufferHandle =
+          nullptr,                          // .FreeBufferHandle =
+          external_delegate_->flags,        // .flags =
```
Comment (jamiecook) [inline tensorflow/lite/delegates/external/external_delegate.cc:163]: As designated initializers are only supported on MSVC 2019 with stdc++20 and this is the only use of the syntax in the codebase, I modified to use the standard syntax and kept the field names as comments for readability.

**Comment by jamiecook** [comment]
@gbaned - I'm not sure what to with the failing CI check - it failed with `Temporary failure resolving 'ppa.launchpad.net'` during a package install during python testing - obviously nothing to do with this PR.

File: tensorflow/lite/kernels/internal/compatibility.h [hunk 2]
```diff
@@ -86,6 +86,17 @@ using int32 = std::int32_t;
 using uint32 = std::uint32_t;
 #endif  // !defined(TF_LITE_STATIC_MEMORY)
 
+
+// Allow for cross-compiler usage of function signatures - used for specifying
```
Comment (jamiecook) [inline tensorflow/lite/kernels/internal/compatibility.h:90]:
```suggestion
// Allow for cross-compiler usage of function signatures - currently used for specifying 
```

**Review by mihaimaruseac — APPROVED** [review]

**Comment by mihaimaruseac** [comment]
MLIR breakage at the point this PR was created.

**Comment by jamiecook** [comment]
Again CI doesn't seem to be building in an unrelated component (python //py_test_dir/tensorflow/python/distribute/failure_handling:failure_handler_test failed in 75% of cases) all other python tests passed: 

`Executed 821 out of 821 tests: 820 tests pass and 1 fails locally.`

image

**Comment by mihaimaruseac** [comment]
Known internal breakage. Test disabled.

At the moment, once required builds passed and import/copybara is also green, you are waiting on internal review. Reviewer with approval rights has not yet acted on the internal version of this PR. You are in the bottom left corner here:

life_of_pr

File: tensorflow/lite/kernels/internal/optimized/depthwiseconv_uint8.h [hunk 3]
```diff
@@ -1519,7 +1519,7 @@ void QuantizedDepthwiseConvAccumRow(int stride, int dilation_factor,
                                     int16 filter_offset, int out_x_buffer_start,
                                     int out_x_buffer_end, int output_depth,
                                     int32* acc_buffer) {
-  ruy::profiler::ScopeLabel label(__PRETTY_FUNCTION__);
+  ruy::profiler::ScopeLabel label(TFLITE_PRETTY_FUNCTION);;
```
Comment (mihaimaruseac) [inline tensorflow/lite/kernels/internal/optimized/depthwiseconv_uint8.h:1522]:
```suggestion
  ruy::profiler::ScopeLabel label(TFLITE_PRETTY_FUNCTION);
```

**Review by mihaimaruseac — APPROVED** [review]

**Comment by mihaimaruseac** [comment]
So internal build was failing because of the double `;;`. Removed one in my commit above, now it's back to internal presubmit, but has been reviewed, so if nothing else happens it should get merged

**Pull request merged**